apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/LocalParticipantWrapper.kt

```diff
@@ -17,157 +18,178 @@ import com.twilio.video.ktx.createLocalAudioTrack
+        localVideoTrack?.let {
+            localParticipant?.publishTrack(it, LocalTrackPublicationOptions(TrackPriority.LOW))?.also {
+                isCameraOn = true
             }
         }
     }
 
-    private fun publishAudioTrack(localAudioTrack: LocalAudioTrack?) {
-        if (!isAudioMuted) {
-            localAudioTrack?.let { participant?.publishTrack(it) }
-        }
-    }
+    private fun unPublishTrack(localVideoTrack: LocalVideoTrack?) = localVideoTrack?.let { localParticipant?.unpublishTrack(it) }
 
-    private fun unPublishTrack(localVideoTrack: LocalVideoTrack?) =
-        localVideoTrack?.let { participant?.unpublishTrack(it) }
+    private fun unPublishTrack(localAudioTrack: LocalAudioTrack?) = localAudioTrack?.let { localParticipant?.unpublishTrack(it) }
 
-    private fun unPublishTrack(localAudioTrack: LocalAudioTrack?) =
-        localAudioTrack?.let { participant?.unpublishTrack(it) }
-
-    private fun removeCameraTrack() {
-        videoTrack?.let { cameraVideoTrack ->
+    private fun removeCameraTrack(isCameraOn: Boolean? = null) {
+        localVideoTrack?.let { cameraVideoTrack ->
             unPublishTrack(cameraVideoTrack)
             localVideoTrackNames.remove(cameraVideoTrack.name)
             cameraVideoTrack.release()
-            this.videoTrack = null
+            localVideoTrack = null
+            isCameraOn?.also {
+                this.isCameraOn = it
+            }
         }
     }
 
-    private fun removeAudioTrack() {
+    private fun removeAudioTrack(isMicOn: Boolean = this.isMicOn) {
         localAudioTrack?.let { localAudioTrack ->
             unPublishTrack(localAudioTrack)
             localAudioTrack.release()
             this.localAudioTrack = null
+            this.isMicOn = isMicOn
         }
     }
 
     override fun onAudioTrackPublished(
         localParticipant: LocalParticipant,
         localAudioTrackPublication: LocalAudioTrackPublication
     ) {
-        //TODO("Not yet implemented")
+        Timber.d("onAudioTrackPublished")
     }
 
     override fun onAudioTrackPublicationFailed(
         localParticipant: LocalParticipant,
         localAudioTrack: LocalAudioTrack,
         twilioException: TwilioException
     ) {
-        //TODO("Not yet implemented")
+        Timber.d("onAudioTrackPublicationFailed ${twilioException.message}")
     }
 
     override fun onVideoTrackPublished(
         localParticipant: LocalParticipant,
         localVideoTrackPublication: LocalVideoTrackPublication
     ) {
-        //TODO("Not yet implemented")
+        Timber.d("onVideoTrackPublished")
     }
 
     override fun onVideoTrackPublicationFailed(
         localParticipant: LocalParticipant,
         localVideoTrack: LocalVideoTrack,
         twilioException: TwilioException
     ) {
-        //TODO("Not yet implemented")
+        Timber.d("onVideoTrackPublicationFailed ${twilioException.message}")
```

Comment:
I assume at a later point you will replace this with a means of percolating the error the user and not just to logcat?

Reply:
Yes, that would be an error flow. I already have an `Error` state to notify to the UI. Good catch, I will take care of it in a different task about optimizing error flows.